src/core/snippets/snippets.ts

```diff
@@ -83,21 +109,42 @@ export class UIPSnippets extends UIPPlugin {
     const $li = document.createElement('li');
     $li.classList.add(UIPSnippets.ITEM_CLASS);
     $li.textContent = label;
-    $li.appendChild(snippet);
     return $li;
   }
 
-  /** Apply active snippet's markup to {@link UIPStateModel}. */
-  protected applyActive(): void {
-    const tmpl = this.$active?.querySelector(UIPSnippets.CONTENT_SEL);
-    if (!tmpl || !this.model) return;
-    this.model.setHtml(tmpl.innerHTML, this);
+  protected updateTitleText(snippet: HTMLTemplateElement) {
+    const titleText = this.querySelector('.snippets-title') as HTMLElement;
+    titleText.textContent = `${snippet.getAttribute('label')}`;
   }
 
   @bind
-  protected _onClick(event: Event) {
+  protected _onListClick(event: Event) {
     const target = event.target as HTMLElement;
-    if (!target.classList.contains(UIPSnippets.ITEM_CLASS)) return;
-    this.$active = target;
+    const index = this.$items.indexOf(target);
+    if (index < 0) return;
+
+    this.toggleDropdown();
+
+    if (UIPSnippets.currentIndex !== index && index >= 0 && this.model) {
```

Comment:
Does this check `index >= 0` reasonable after upper `if (index < 0) return;` ?

Reply:
Missed it) Fixed